File: my-folder/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = defaultdict(list)

        for i, num in enumerate(nums):
            table[num].append(i)
        
        ans = []

        for i, num in enumerate(nums):
            second = target - num
            if second in table:
                if second == num:
                    if len(table[second]) > 1:
                        ans.append(i)
                        ans.append(table[num].pop())
                        break
                else:
                    ans.append(i)
                    ans.append(table[second].pop())
                    break
                
        return ans

File: my-folder/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> table;
        
        for (int i = 0; i < nums.size(); i++){
            int second = target - nums[i];
            if (table.find(second) != table.end() && table[second]!=i)
                return {table[second], i};
            table[nums[i]] = i;

        }
        return {};
    }
};
